artemis-core-client/src/main/java/org/apache/activemq/artemis/core/server/QueueQueryResult.java

```diff
@@ -55,13 +57,19 @@ public QueueQueryResult(final SimpleString name,
                            final int consumerCount,
                            final long messageCount,
                            final boolean autoCreateQueues,
+                           final boolean autoCreateQueuesDurable,
                            final boolean exists,
                            final boolean autoCreated,
                            final boolean purgeOnNoConsumers,
                            final RoutingType routingType,
                            final int maxConsumers) {
+
       this.durable = durable;
 
+      if (autoCreateQueues) {
+         this.durable = autoCreateQueuesDurable;
```

Comment:
As noted this should be an enum, and only override, if enum is not null, currently his just overrides the value with no option to not override.